libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -571,6 +600,68 @@ bool AP_BattMonitor::reset_remaining(uint16_t battery_mask, float percentage)
     return ret;
 }
 
+// Logs smart battery messages
+void AP_BattMonitor::log_smart_battery_info() const
+{    
+    for(uint8_t i = 0; i < _num_instances; i++) {
+        if (get_type(i) > AP_BattMonitor::Type::ANALOG_VOLTAGE_AND_CURRENT) {
+            log_smart_battery_info(i);
+        }
+    }
+}
+
+void AP_BattMonitor::log_smart_battery_info(uint8_t instance) const
+{
+    uint16_t cycles;
+    bool got_cycle_count = get_cycle_count(instance,cycles);
+
+    int32_t capacity_full = pack_capacity_mah(instance); //battery._params->_pack_capacity
+
+    if (capacity_full == 0) {
+        capacity_full = -1;
+    }
+
+    char serial_number[MAVLINK_MSG_SMART_BATTERY_INFO_FIELD_SERIAL_NUMBER_LEN]{};
+    get_serial_number(instance, serial_number, ARRAY_SIZE(serial_number)); 
+
+    char product_name[MAVLINK_MSG_SMART_BATTERY_INFO_FIELD_DEVICE_NAME_LEN]{}; 
+    get_product_name(instance, product_name, ARRAY_SIZE(product_name));
+
+    // @LoggerMessage: BATI
+    // @Vehicles: All
+    // @Description: Smart Battery Information
+    // @Field: TimeUS: Time since system startup
+    // @Field: id: battery monitor instance number
+    // @Field: func: battery function
+    // @Field: type: battery type (chemistry)
+    // @Field: capD: design capacity when full according to manufacturer, -1: field not provided.
+    // @Field: capF: capacity when full (accounting for battery degradation), -1: field not provided.
+    // @Field: cyc: charge/discharge cycle count. UINT16_MAX: field not provided.
+    // @Field: ser: serial number in ASCII characters, 0 terminated. All 0: field not provided.
+    // @Field: name: product name encoded as 'device names'_'manufacturer'
+    // @Field: w: [newtons] battery weight. 0: field not provided. 
+    // @Field: dminv: minimum per-cell voltage when discharging. If not supplied set to UINT16_MAX value.  
+    // @Field: cminv: minimum per-cell voltage when charging. If not supplied set to UINT16_MAX value.  
+    // @Field: rminv: minimum per-cell voltage when resting. If not supplied set to UINT16_MAX value. 
+    AP::logger().Write("BATI", "TimeUS,id,func,type,capD,capF,cyc,ser,name,w,dminv,cminv,rminv",
+                       "s#--ii?--Nvvv",
+                       "F000!!0--C000",
+                       "QBBBiiHNZHHHH",
+                        AP_HAL::micros64(),
+                        instance,
+                        MAV_BATTERY_FUNCTION_UNKNOWN,
+                        MAV_BATTERY_TYPE_UNKNOWN,
+                        -1,
+                        capacity_full,
+                        got_cycle_count ? cycles : UINT16_MAX,
+                        serial_number,
+                        product_name,
+                        0, // Newtons = 9.81*1E-3*g                 
+                        UINT16_MAX,
+                        UINT16_MAX,
+                        UINT16_MAX);
```

Comment:
This a low-bandwidth message.  We could just use floats here and use a proper flag value - `NaN` - to say we don't know.

Reply:
Fixed